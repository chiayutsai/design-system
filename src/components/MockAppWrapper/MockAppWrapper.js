import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
