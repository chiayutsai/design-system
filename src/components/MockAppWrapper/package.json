{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
