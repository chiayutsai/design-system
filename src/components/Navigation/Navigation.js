/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import classnames from 'classnames'
import useStyles from 'isomorphic-style-loader/useStyles'
import Link from '../Link'
import styles from './Navigation.scss'

const Navigation = () => {
  useStyles(styles)
  return (
    <div className={styles.root} role="navigation">
      <Link className={styles.link} to="/about">
        About
      </Link>
      <Link className={styles.link} to="/contact">
        Contact
      </Link>
      <span className={styles.spacer}> | </span>
      <Link className={styles.link} to="/login">
        Log in
      </Link>
      <span className={styles.spacer}>or</span>
      <Link
        className={classnames(styles.link, styles.highlight)}
        to="/register">
        Sign up
      </Link>
    </div>
  )
}

export default Navigation
