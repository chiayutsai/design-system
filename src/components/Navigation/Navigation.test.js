import renderer from 'react-test-renderer'
import MockAppWrapper from 'components/MockAppWrapper'
import Navigation from './Navigation'

describe('Navigation', () => {
  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <Navigation />
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
