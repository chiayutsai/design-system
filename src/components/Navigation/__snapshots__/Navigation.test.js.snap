// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation render correctly 1`] = `
<div
  className="root"
  role="navigation"
>
  <a
    className="link"
    href="/about"
    onClick={[Function]}
  >
    About
  </a>
  <a
    className="link"
    href="/contact"
    onClick={[Function]}
  >
    Contact
  </a>
  <span
    className="spacer"
  >
     | 
  </span>
  <a
    className="link"
    href="/login"
    onClick={[Function]}
  >
    Log in
  </a>
  <span
    className="spacer"
  >
    or
  </span>
  <a
    className="link highlight"
    href="/register"
    onClick={[Function]}
  >
    Sign up
  </a>
</div>
`;
